web/app/themes/mitlib-parent/js/alerts.js

```diff
@@ -0,0 +1,127 @@
+// Loads alert-level posts on the top of all pages
+function filterAlerts(posts) {
+	// This processes an array of posts for valid, confirmed, alerts
+	var filtered = [],
+		post,
+		post_meta,
+		i;
+
+	// For each post...
+	for (i = 0; i < posts.length; i++) {
```

Comment:
It does loop over all posts on every page load - but the number of posts is pretty small (right now there is only one visible at https://libraries.mit.edu/wp-json/wp/v2/posts/ ).

The other conditions are because not every Post record is necessarily an Alert, and also because there is a safety field that must be checked in order to it to really be displayed here. I would like, ultimately, to make Alerts into their own post type - which would IMO allow us to avoid all of this - but we decided a few weeks ago to avoid this work for now.